mvlearn/split/split_transformers.py

```diff
@@ -0,0 +1,221 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# Authors: Pierre Ablin
+
+import numpy as np
+
+from abc import abstractmethod
+
+from sklearn.base import TransformerMixin
+from sklearn.utils import check_array
+from sklearn.utils.validation import check_is_fitted
+
+from ..utils.utils import check_Xs
+
+
+class BaseSplitter(TransformerMixin):
+    """A base class for splitting single view datasets into multiview datasets
+
+    The .transform function should return a multiview dataset
+
+    Parameters
+    ----------
+    Attributes
+    ----------
+    See Also
+    --------
+    """
+    def __init__(self):
+        pass  # pragma: no cover
+
+    @abstractmethod
+    def fit(self, X, y=None):
+        r"""Fit model to multiview data.
+
+        Parameters
+        ----------
+        X: array of shape (n_samples, n_features)
+            Input data
+
+        y : array, shape (n_samples,), optional
+
+        Returns
+        -------
+        self: returns an instance of self.
+        """
+
+        return self  # pragma: no cover
+
+    @abstractmethod
+    def transform(self, X, y=None):
+        r"""Split singleview dataset into multiple views
+
+        Parameters
+        ----------
+        X: array of shape (n_samples, n_features)
+            Input data
+
+        y : array, shape (n_samples,), optional
+
+        Returns
+        -------
+        Xs_transformed : list of array-likes
+            - Xs shape: (n_views,)
+            - Xs[i] shape: (n_samples, n_features_i)
+        """
+        pass  # pragma: no cover
+
+    def fit_transform(self, X, y=None):
+        r"""Fit to the data and split
+
+        Parameters
+        ----------
+        X: array of shape (n_samples, n_features)
+            Input data
+
+        y : array, shape (n_samples,), optional
+
+        Returns
+        -------
+        Xs_transformed : list of array-likes
+            - Xs shape: (n_views,)
+            - Xs[i] shape: (n_samples, n_features_i)
+        """
+        return self.fit(X, y).transform(X)
+
+    @abstractmethod
+    def inverse_transform(self, Xs):
+        r"""Take a multiview dataset and merge it into a single view dataset
+
+        Parameters
+        ----------
+        Xs : list of numpy.ndarray
+            - Xs length: n_views
+            - Xs[i] shape: (n_samples, n_features_i)
+
+        Returns
+        -------
+        X : numpy.ndarray, shape (n_total_features, n_samples)
+            The input dataset
+
+        """
+
+        pass  # pragma: no cover
+
+
+class ConcatSplitter(BaseSplitter):
```

Comment:
the BaseSplitter class is an overkill until you have 2 splitter classes

Reply:
Even if we add some other classes later on? for instance we can add a class that repeats a single dataset.

Reply:
Base class has been written, removing it seems like a waste.

Reply:
do things when you need them. 

https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it

you can always find the code in the PR or git history.

I've seen too much code written and maintained because "we will surely need it later"...